### DISCUSSION TITLE
searchBar autoFocus

### DISCUSSION THREAD

**Discussion by cabiad** [software-mansion/react-native-screens#1007] [Q&A] [answered]
A normal react native [TextInput has an `autoFocus` property](https://reactnative.dev/docs/textinput#autofocus). Is there a way to have the search field automatically focused when the screen is focused with the current searchBar implementation?

I feel like I may be missing something obvious, but I didn't see it a way to do it in the docs or in the exposed APIs [here](https://github.com/software-mansion/react-native-screens/blob/e02172a08b694bd9a8baf7e94c21124b195bd268/ios/RNSSearchBar.m). (I see an `onFocus()` event, but no way to automatically or programmatically focus the search field.)

**Comment by tboba** [comment, marked as answer]
Hi @cabiad, sorry for the late response 🙏 
It is! There's an `autoFocus` field which controls if the search bar should be autofocused on screen mount.

Cheers!
